### ISSUE TITLE
User Profile page in NextJS

### ISSUE THREAD

**Issue by rajraj889** [Auto-DL/Auto-DL#384]
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

New Implementation

### Feature Description

![image](https://user-images.githubusercontent.com/40288700/135686334-b1a6b4fc-fa21-4884-bf59-8bcd0ddeca94.png)
Profile section needed for user, where he/she can edit some details and add a profile pic

### Possible Solution

Profile page where user can see all his/her info and edit details like password, first name, last name.
Username, email should only be read only for now.

**Comment by precious-adeyinka** [comment]
Hey @rajraj889 , do you need me to figure out the UI also and also develop the UI components only?

**Comment by rajraj889** [comment]
@precious-adeyinka would love that, thats for your interest in the project :) we use figma for designing in this project.if you can do some prototyping in figma would be great.

**Comment by HarshitPandey251** [comment]
@rajraj889 I want to work on it

**Comment by McTechie** [comment]
Heyy @HarshitPandey251 as this issue is already assigned, it would be great if you could tackle some other ones for now!

**Comment by precious-adeyinka** [comment]
Hey @rajraj889 and @McTechie what do you think about this profile design, it is very minimalist I know, but can we proceed with this?

**Check out the figma design here** - https://www.figma.com/file/Yv9YeTT3770271rd1wqwce/Auto-DL-Profile-Page?node-id=0%3A1

**Comment by precious-adeyinka** [comment]
Okay, so it looks like this.

![image](https://user-images.githubusercontent.com/20630901/135722545-bd27ab4e-d7b2-417b-9e71-b13c93cbf34e.png)

**Comment by RusherRG** [comment]
This looks good for now I guess. I like it. You can follow the colourscheme mentioned in `FrontEndApp/autodl-platform/styles/theme.tsx`

**Comment by precious-adeyinka** [comment]
Okay, @RusherRG I will check that out.

**Comment by precious-adeyinka** [comment]
I am heading out briefly, when I am back I will continue.

**Comment by rajraj889** [comment]
@precious-adeyinka Yes the design looks good.

**Comment by precious-adeyinka** [comment]
@RusherRG this is what I have using the color palette in the `FrontEndApp/autodl-platform/styles/theme.tsx`

![image](https://user-images.githubusercontent.com/20630901/135725553-733632ca-4f85-41de-b0c9-f72cb76e9d60.png)

**Comment by precious-adeyinka** [comment]
I see that you are currently using material-ui, I wanted to know if I could use tailwindcss for the profile page, and also confirm that the changes are to be made in the` autodl-platform` folder yeah?

**Comment by RusherRG** [comment]
Yeah you have to make the changes in `FrontEndApp/autodl-platform`. I think it's better if we stick to material-ui. Is there any specific reason why you want to use tailwind?

**Comment by RusherRG** [comment]
> @RusherRG this is what I have using the color palette in the `FrontEndApp/autodl-platform/styles/theme.tsx`
> 
> ![image](https://user-images.githubusercontent.com/20630901/135725553-733632ca-4f85-41de-b0c9-f72cb76e9d60.png)

This looks pretty nice. Go ahead and implement the components

**Comment by precious-adeyinka** [comment]
Yeah I don't use material-ui normally, I am more productive with tailwindcss or vanilla css, but if you don't have any problem with an extended deadline, I could look into the material-ui docs, but that's a whole new learning journey before implementation, but I will try to make it fast as I can.

**Comment by precious-adeyinka** [comment]
Let me know what you decide upon, so I can proceed further.

**Comment by RusherRG** [comment]
The material-ui docs are pretty easy to understand with sample codes. We'll like for you to use material-ui itself.

> ut if you don't have any problem with an extended deadline,

Yeah absolutely... not an issue. Just mention an appropriate deadline and you could probably start working on it. Thanks a lot for the design, would love to see it implemented.

**Comment by precious-adeyinka** [comment]
Awesome, no worries, less than a week or about a week would be perfect okay.

**Comment by precious-adeyinka** [comment]
Hello guys, slight update. I have submitted a PR and I am done with the UI - https://github.com/Auto-DL/Auto-DL/pull/414

**Comment by precious-adeyinka** [comment]
This is how it looks now on my dev env!

![image](https://user-images.githubusercontent.com/20630901/135952393-2ec79b65-39f4-4edc-8a1f-306d163d5784.png)

**Comment by precious-adeyinka** [comment]
Everything is working perfectly!
